src/ui/hooks/showPassword/index.ts

```diff
@@ -0,0 +1,19 @@
+import { useState } from "react";
+
+export interface IUseShowPassword {
+  isShowPassword: boolean;
+  setShowPassword: () => void;
+}
+
+export const useShowPassword = (): IUseShowPassword => {
+  const [isShowPassword, setIsShowPassword] = useState(false);
+
+  const setShowPassword = () => {
```

Comment:
Please, add `useCallback` with `setIsShowPassword` and rename it to `onShowPassword`.